src/VCR/Util/StreamProcessor.php

```diff
@@ -258,7 +258,11 @@ public function stream_seek($offset, $whence = SEEK_SET)
      */
     public function stream_stat()
     {
-        return fstat($this->resource);
+        $this->restore();
+        $stat = fstat($this->resource);
+        $this->intercept();
```

Comment:
any idea why this messes up the tests as much as it does?

anyways, not very sure about this change. but without it i got weird errors too